static-site/src/sections/group-members-page.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import { uri } from "../../../src/templateLiterals.js";
+import GenericPage from "../layouts/generic-page.jsx";
+import { BigSpacer, CenteredContainer, FlexGridRight, MediumSpacer } from "../layouts/generalComponents.jsx";
+import * as splashStyles from "../components/splash/styles";
+import { ErrorBanner } from "../components/errorMessages.jsx";
+
+interface GroupMember {
+  username: string,
+  roles: string[]
+}
+
+interface ErrorMessage {
+  title: string,
+  contents: string
+}
+
+const GroupMembersPage = ({ groupName }: {groupName: string}) => {
+  const [ errorMessage, setErrorMessage ] = useState<ErrorMessage>({title: "", contents: ""});
+  const [ roles, setRoles ] = useState<string[]>([]);
+  const [ members, setMembers ] = useState<GroupMember[]>([]);
+
+  useEffect(() => {
+    async function getGroupMembership(groupName: string) {
+      const headers = { headers: {"Accept": "application/json"}};
+      let roles, members = [];
+      try {
+        const [ rolesResponse, membersResponse ] = await Promise.all([
+          fetch(uri`/groups/${groupName}/settings/roles`, headers),
+          fetch(uri`/groups/${groupName}/settings/members`, headers)
+        ]);
+        if (!rolesResponse.ok) {
+          throw new Error(`Fetching group roles failed: ${rolesResponse.status} ${rolesResponse.statusText}`)
+        }
+        if (!membersResponse.ok) {
+          throw new Error(`Fetching group members failed: ${membersResponse.status} ${membersResponse.statusText}`)
+        }
+        roles = await rolesResponse.json();
+        members = await membersResponse.json();
+      } catch (err) {
+        const errorMessage = (err as Error).message
+        if(!ignore) {
+          setErrorMessage({
+            title: "An error occurred when trying to fetch group membership data",
+            contents: errorMessage})
+        }
+      }
+      return {roles, members};
+    }
+
+    let ignore = false;
+    getGroupMembership(groupName).then(result => {
+      if (!ignore) {
+        setRoles(result.roles);
+        setMembers(result.members);
+      }
+    })
+    return () => {
+      ignore = true;
+    };
+  }, [groupName]);
+
+  return (
+    <GenericPage banner={errorMessage.title ? <ErrorBanner {...errorMessage} /> : undefined}>
+      <FlexGridRight>
+        <splashStyles.Button to={uri`/groups/${groupName}`}>
+          Return to {`"${groupName}"`} Page
+        </splashStyles.Button>
+      </FlexGridRight>
+      <MediumSpacer/>
+
+      <splashStyles.H2>
+        {`"${groupName}"`} Group Members
+      </splashStyles.H2>
+      <BigSpacer/>
+
+      {roles && members
+        ? <MembersTable roles={roles as string[]} members={members as GroupMember[]} />
+        : <splashStyles.H4>Fetching group members...</splashStyles.H4>}
+    </GenericPage>
+  )
+};
+
+const MembersTable = ({ roles, members }: { roles: string[], members: GroupMember[]}) => {
```

Comment:
Current spacing makes it hard to visually link username to the role which is far to the right, especially for a long list of members. Consider adding horizontal lines between rows or borders on the entire table to add visual guides across the whitespace.

Reply:
Thanks for the push here, I add more styling in https://github.com/nextstrain/nextstrain.org/pull/973/commits/fe814eef7e44f162515a692572c482fc159db4f8

![Screenshot 2024-08-09 at 4 30 18 PM](https://github.com/user-attachments/assets/eee78b1a-93e7-47fa-8f1f-bc28ec5358de)